# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/l10n/DOMLocalization.h
# Commit: 85194e5b7a9e
# Full Hash: 85194e5b7a9e6ed9114ee46aef0b5efb819cca9b
# Author: Zibi Braniecki <zbraniecki@mozilla.com>
# Date: 2021-08-04 03:56:33
# Regressor Bug: 1613705
# File Overlap Count: 1
# Description:
#   Bug 1613705 - [localization] part9: Update DOMLocalization to match Localization constructor. r=nika,emilio
#   
#   Depends on D113691
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113692
# ==============================================================================

diff -r 181487bdf6c0 -r 85194e5b7a9e dom/l10n/DOMLocalization.h
--- a/dom/l10n/DOMLocalization.h	Tue Aug 03 16:25:13 2021 +0000
+++ b/dom/l10n/DOMLocalization.h	Tue Aug 03 16:25:13 2021 +0000
@@ -15,6 +15,7 @@
 #include "mozilla/dom/L10nOverlaysBinding.h"
 #include "mozilla/dom/LocalizationBinding.h"
 #include "mozilla/dom/PromiseNativeHandler.h"
+#include "mozilla/intl/L10nRegistry.h"
 
 // XXX Avoid including this here by moving function bodies to the cpp file
 #include "nsINode.h"
@@ -30,15 +31,13 @@
   NS_DECL_ISUPPORTS_INHERITED
   NS_DECL_CYCLE_COLLECTION_CLASS_INHERITED(DOMLocalization, Localization)
 
-  static already_AddRefed<DOMLocalization> Create(
-      nsIGlobalObject* aGlobal, const bool aSync,
-      const BundleGenerator& aBundleGenerator);
-
   void Destroy();
 
   static already_AddRefed<DOMLocalization> Constructor(
-      const GlobalObject& aGlobal, const Sequence<nsString>& aResourceIds,
-      const bool aSync, const BundleGenerator& aBundleGenerator,
+      const dom::GlobalObject& aGlobal,
+      const dom::Sequence<nsCString>& aResourceIds, bool aIsSync,
+      const dom::Optional<dom::NonNull<intl::L10nRegistry>>& aRegistry,
+      const dom::Optional<dom::Sequence<nsCString>>& aLocales,
       ErrorResult& aRv);
 
   virtual JSObject* WrapObject(JSContext* aCx,
@@ -112,9 +111,11 @@
     return false;
   }
 
+  DOMLocalization(nsIGlobalObject* aGlobal, bool aSync);
+  DOMLocalization(nsIGlobalObject* aGlobal, bool aIsSync,
+                  const intl::ffi::LocalizationRc* aRaw);
+
  protected:
-  explicit DOMLocalization(nsIGlobalObject* aGlobal, const bool aSync,
-                           const BundleGenerator& aBundleGenerator);
   virtual ~DOMLocalization();
   void OnChange() override;
   void DisconnectMutations();